session.go

```diff
@@ -836,8 +838,12 @@ func (s *session) PrepareStmt(sql string) (stmtID uint32, paramCount int, fields
 		return
 	}
 
+	s.PrepareTxnCtx(goCtx)
 	prepareExec := executor.NewPrepareExec(s, executor.GetInfoSchema(s), sql)
 	err = prepareExec.DoPrepare()
+	if err == nil {
+		err = s.doCommit(goCtx)
```

Comment:
Why `doCommit` here?
What's behavior of MySQL?  

```
begin;
insert into t values (1);
prepare stmt from 'select * from t';
rollback;
```

Will 1 be inserted?

Reply:
fixed.